### ISSUE TITLE
Errors in mnist_tutorial_keras.py

### ISSUE THREAD

**Issue by EmpRamses** [tensorflow/cleverhans#998]
***The issue tracker should only be used to report bugs or feature requests. If you are looking for support from other library users, please ask a question on [StackOverflow](https://stackoverflow.com/questions/tagged/cleverhans).***

**Describe the bug**
For now, I have found two errors in mnist_tutorial_keras.py.
The first error occurs in the original code in the repo.
ValueError: Could not interpret optimizer identifier: <tensorflow.python.keras.optimizers.Adam object at 0x7fa852729f28>
It may be caused by different versions of Keras libraries from Keras-base and keras in Tensorflow.
I have tried to install keras-2.1.6 as Tensorflow has. But it doesn't work.

I tried to import keras without Tensorflow as "import keras". But it causes the second error.
AttributeError: Layer sequential_1 has multiple inbound nodes, hence the notion of "layer input" is ill-defined. Use `get_input_at(node_index)` instead.
This error occurs at input method of Layer class in keras/engine/base_layer.py
I have no idea what caused this error.

Besides, I tried to remove keras library. Then I found that the keras library is required by CleverHans.

**System configuration**
 - CentOS 7
 - Python 3.6.8
 - TensorFlow r1.12
 - Keras 2.2.4

**Comment by EmpRamses** [comment]
I think I have figure out the first error.
First, I import keras directly rather than from tensorflow at the 14th line.
Then, I comment the 87th line which is called to deal with some errors when the model uses custom loss.

These operations only make the traditional model training processes work. But the second error still occurs in the adversarial training.

I have tried to locate the error. I found that Cross-entropy on the adversarial examples in function get_adversarial_loss (line 200) may causes the error. But I still have no idea how to deal with it.
